### PR TITLE
38 get campaign

### PR DESCRIPTION
**Pull request by DeenaAlghazali**

### CODE REVIEW

File: server/controllers/user/getCampaignById.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import { CustomedError } from '../../utils';
+import { paramsSchema } from '../../utils/validation';
+
+const getCampaignById = async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    const result = await paramsSchema.validateAsync(req.params);
+    const { id } = result;
+    const campaignInfo = await Campaign.findByPk(id);
+    if (!campaignInfo) throw new CustomedError('there is no campaign', 400);
+    res.status(200).json({ message: 'campaign information', data: campaignInfo });
```
Comment (Mohammed-Hamada) [inline server/controllers/user/getCampaignById.ts:12]: `Success` because response successfully sent

File: server/tests/route.test.ts [hunk 2]
```diff
@@ -166,6 +166,27 @@ describe('Get/logout', () => {
...
+      target: 50000,
+      is_available: true,
+    };
+    const response = await request(app)
+      .get(`/api/campaign/${id}`)
+      .expect(200);
+    expect(response.body.data.id).toBe(data.id);
+    expect(response.body.data.title).toBe(data.title);
+    expect(response.body.data.description).toBe(data.description);
+    expect(response.body.data.target).toBe(data.target);
```
Comment (Mohammed-Hamada) [inline server/tests/route.test.ts:185]: Format ☺

**Review by Mohammed-Hamada — CHANGES_REQUESTED** [review]

File: server/tests/route.test.ts [hunk 3]
```diff
@@ -166,6 +166,27 @@ describe('Get/logout', () => {
...
+      description: 'This campaign helps save an amount of money that guarantees 50 families for two months',
+      target: 50000,
+      is_available: true,
+    };
+    const response = await request(app)
+      .get(`/api/campaign/${id}`)
+      .expect(200);
+    expect(response.body.data.id).toBe(data.id);
+    expect(response.body.data.title).toBe(data.title);
+    expect(response.body.data.description).toBe(data.description);
```
Comment (Raghad-Mezied) [inline server/tests/route.test.ts:184]: Why not test them in just one line, I mean you have `response.body` and it should equal what you want.
And test the other cases.

File: server/routes/userRoutes.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import { Router } from 'express';
+import { getCampaignById } from '../controllers';
+
+const userRouter = Router();
+
+userRouter.route('/campaign/:id').get(getCampaignById);
```
Comment (Raghad-Mezied) [inline server/routes/userRoutes.ts:6]: Name it `compaignRouter`

File: server/controllers/user/getCampaignById.ts [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import { Request, Response, NextFunction } from 'express';
+import { Campaign } from '../../database/models';
+import { CustomedError } from '../../utils';
+import { paramsSchema } from '../../utils/validation';
+
+const getCampaignById = async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    const result = await paramsSchema.validateAsync(req.params);
+    const { id } = result;
```
Comment (Raghad-Mezied) [inline server/controllers/user/getCampaignById.ts:9]: Why not destructure the id immediately? instead of getting the result and then destructuring

**Review by Raghad-Mezied — CHANGES_REQUESTED** [review]

**Review by Raghad-Mezied — APPROVED** [review]

**Review by HaniOlwan — APPROVED** [review]

**Review by Mohammed-Hamada — APPROVED** [review]
Great Work Deena

**Review by farah2003 — APPROVED** [review]

**Pull request merged**